### PR TITLE
Use Round Token to Calculate Matches, Min amount, Sybil Defense, Matching Cap

### PR DESCRIPTION
**Pull request by boudra**
resolves https://github.com/gitcoinco/allo-indexer/issues/40

It also implements:

- minimum amount
- passport option
- matching cap

merge this together: https://github.com/gitcoinco/grants-stack/pull/1659

### CODE REVIEW

**Comment by boudra** [comment]
@vacekj @hmrtn I deleted a duplicate test, please let me know if this is wrong, check another comment here: https://github.com/gitcoinco/allo-indexer/pull/49/files#r1182228669

File: package.json [hunk 1]
```diff
@@ -40,7 +40,7 @@
     "express": "^4.18.2",
     "multer": "^1.4.5-lts.1",
     "npm-run-all": "^4.1.5",
-    "pluralistic": "github:gitcoinco/pluralistic.js#431f4fe2d429f65f2de03bdfcc66bba451da2ba7",
+    "pluralistic": "github:gitcoinco/pluralistic.js#8f7c85f1f0ddbe91ed849e9f21a5a80c0f05916a",
```
Comment (gravityblast) [inline package.json:43]: we need to change it to the latest where we fix a bug

**Review by hmrtn — APPROVED** [review]
lgtm!

File: package.json (hunk 1 above)
Comment (boudra) [reply package.json]: done

**Review by gravityblast — APPROVED** [review]

**Pull request merged**